client/aws_credentials.go

```diff
@@ -40,7 +40,6 @@ func (self *ApiClient) AwsCredentialsCreate(request AwsCredentialsCreatePayload)
 		return ApiKey{}, err
 	}
 
-	request.Type = "AWS_ASSUMED_ROLE_FOR_DEPLOYMENT"
```

Comment:
It could break the usage of customers.

I agree this default is ugly - but if someone uses this resource. it will force him to commit change before new deploy. 

maybe use the default with warning message

Reply:
@HeverFarber but I think it is still backward's compatible... the old usage still requires `arn`& `externa_id`, and the `Type` was assigned under the hood.

Reply:
@HeverFarber  as @liranfarage89 said Its backward's compatible.

Reply:
We should keep backward's compatible. for the who doesn't add type before. we can add it instead of him

Reply:
but the type was not exposed before that...

Reply:
Correct - so we added it for the user. so the user didn't add it to his resource.

Now when we removed it - the resource without this type - and when we will send it to the server we will get an error because type is missing

Reply:
the Type was not a part of the resource either way, it's computed

Reply:
I don't think the user will need to approve a change since the `Type` isn't part of the schema,
and the removal is in the `client` but @GiliFaroEnv0 is calculating the type on runtime in the "resource" side and sends it to the `client` here 

https://github.com/env0/terraform-provider-env0/blob/0ea2c1771e18d41948de435cd344a0ebe548b868/env0/resource_aws_credentials.go#L72-L92

and you guys can check it, run `terraform apply` using `main`, then checkout this branch and run `apply` again and check if there is any change

Reply:
OK it works I merge IT 👍🏿 
bors r+